lambdas/src/apis/collections/controllers/collections.ts

```diff
@@ -11,92 +14,139 @@ import {
+    )
+
+    const tagTraits = wearableMetadata.data.tags.reduce((tags: ERC721StandardTrait[], tag) => {
+      tags.push({
+        trait_type: 'Tag',
+        value: tag
+      })
+
+      return tags
+    }, [])
+
+    const standardErc721 = {
+      id: urn,
+      name,
+      description,
+      language: 'en-US',
+      image,
+      thumbnail,
+      attributes: [
+        {
+          trait_type: 'Rarity',
+          value: wearableMetadata.rarity
+        },
+        {
+          trait_type: 'Category',
+          value: wearableMetadata.data.category
+        },
+        ...tagTraits,
+        ...bodyShapeTraits
+      ]
     }
+    res.send(standardErc721)
   } catch (e) {
-    res.status(500).send(e.message)
+    res.status(e.statusCode ?? 500).send(e.message)
   }
 }
 
-export async function contentsImage(client: SmartContentClient, req: Request, res: Response): Promise<void> {
+export async function contentsThumbnail(
+  client: SmartContentClient,
+  req: Request,
+  res: Response,
+  storage: string
+): Promise<void> {
   // Method: GET
-  // Path: /contents/:urn/image
-  const { urn } = req.params
-  await internalContents(client, res, urn, (wearableMetadata) => wearableMetadata.image)
+  // Path: /contents/:urn/thumbnail/:size
+  await handleImageRequest(client, req, res, storage, true)
 }
 
-export async function contentsThumbnail(client: SmartContentClient, req: Request, res: Response): Promise<void> {
+export async function contentsImage(
+  client: SmartContentClient,
+  req: Request,
+  res: Response,
+  storage: string
+): Promise<void> {
   // Method: GET
-  // Path: /contents/:urn/thumbnail
+  // Path: /contents/:urn/image/:size
+  await handleImageRequest(client, req, res, storage, false)
+}
+
+async function handleImageRequest(
+  client: SmartContentClient,
+  req: Request,
+  res: Response,
+  rootStorageLocation: string,
+  isThumbnail: boolean
+): Promise<void> {
   const { urn } = req.params
+  const size = req.params.size ?? DEFAULT_IMAGE_SIZE
+
+  try {
+    validateSize(size)
+
+    const entity = await fetchEntity(client, urn)
+    const hash = getFileHash(entity, entity.metadata.thumbnail)
 
-  await internalContents(client, res, urn, (wearableMetadata) => wearableMetadata.thumbnail)
+    await pruneObsoleteImages(rootStorageLocation, urn, hash)
+
+    const imageRequest = {
+      urn,
+      hash,
+      size,
+      rarityBackground: isThumbnail ? undefined : entity.metadata?.rarity
+    }
+
+    const image = await getImage(client, rootStorageLocation, imageRequest)
+    sendImage(res, image, imageRequest)
+  } catch (e) {
+    res.status(e.statusCode ?? 500).send(e.message)
+  }
+}
+
+function sendImage(res: Response, image: Buffer, imageRequest: ImageRequest) {
+  res.send(image)
+  res.writeHead(200, {
+    'Content-Type': 'image/png',
+    ETag: JSON.stringify(getImageRequestId(imageRequest)),
+    'Access-Control-Expose-Headers': '*',
+    'Cache-Control': 'public, max-age=31536000, immutable'
```

Comment:
Add `s-maxage` to enable edge & cdn caching
```suggestion
    'Cache-Control': 'public, max-age=31536000, s-maxage=31536000, immutable'
```